### PR TITLE
[Feat] auto close issues/pr and auto merge branches

### LINKED ISSUE #67: [feat]: Create workflow that closes pull request and auto merge branches
### Feature description

Create workflow that closes pull request/Issues and auto merge branches

### Related Code

_No response_

### Additional Context

_No response_

### Self-service

- [ ] I'd be willing to address this documentation request myself.

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/srealmoreno/rae-tesis/blob/main/docs/CONTRIBUTING.md).

### Are you familiar with Contributor Covenant Code of Conduct?

- [X] I have read the [Contributor Covenant Code of Conduct](https://github.com/srealmoreno/rae-tesis/blob/main/docs/CODE_OF_CONDUCT.md).

### Contact Details

_No response_

### PR DESCRIPTION
**Pull request by srealmoreno**
# Auto close issues/pull request and auto merge branches

Resolve #67 

## Type of changes

- [x] CI
- [ ] Bugfix
- [x] Feature
- [ ] Generic maintenance tasks
- [ ] Documentation content changes
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no URL changes)
- [ ] Improving performance of the project (not introducing new features)
- [ ] Other (please describe):

Issue Number: #67 

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

N/A

### CODE REVIEW

File: .github/workflows/03-ci-rebase-squash.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          body: |
+            ### Permission denied ⛔
+
+            @${{ github.event.comment.user.login }} this functionality is not available to you.
+            To use this functionality you must meet at least one of the following requirements:
+
+            - [ ] Be the owner  of the repository
+            - [ ] Be a   member of the organization
+            - [ ] Be on the assignees list
+            - [ ] Be the author of the pull request
```
Comment (srealmoreno) [inline .github/workflows/03-ci-rebase-squash.yml:131]: ### Rebase or squash branch

#### Description

rebase or squash current branch (main->current) automatically with a comment in the Pull request.

#### Usage

comment: "@gitbot rebase"  for rebase branch
comment: "@gitbot squash"  for squash branch

#### Requirements

The author of the comment must meet at least one requirement:

- [ ] Be the owner of the repository
- [ ] Be a member of the organization
- [ ] Be on the assignees list
- [ ] Be the author of the commit

#### Interactions

 👀. Caught comment:

  1. When the comment is detected (e.g. @gitbot rebase) the reaction '👀' will be added to the comment.

 👎. Failed:

  1. If the author doesn't meet the requirements, the reaction '👎' will be added to the comment and the comment will be answered indicating which has no necessary requirements.
  2. If the autor meets requirements and the action fails, the reaction '👎' will be added to the comment.

 👍. Success:

  1. If the author meets requirements and the action is executed successfully, the reaction '👍' will be added to the comment.

**Comment by srealmoreno** [comment]
[Add 03-ci-rebase-squash.yml](https://github.com/srealmoreno/rae-tesis/pull/85/commits/ded6b95eeefd9bbe04ab9c7c42fd621f5a284c41)

### Rebase or squash branch

#### Description

rebase or squash current branch (main->current) automatically with a comment in the Pull request.

#### Usage

comment: "@gitbot rebase"  for rebase branch
comment: "@gitbot squash"  for squash branch

#### Requirements

The author of the comment must meet at least one requirement:

- [ ] Be the owner of the repository
- [ ] Be a member of the organization
- [ ] Be on the assignees list
- [ ] Be the author of the commit

#### Interactions

 👀. Caught comment:

  1. When the comment is detected (e.g. @gitbot rebase) the reaction '👀' will be added to the comment.

 👎. Failed:

  1. If the author doesn't meet the requirements, the reaction '👎' will be added to the comment and the comment will be answered indicating which has no necessary requirements.
  2. If the autor meets requirements and the action fails, the reaction '👎' will be added to the comment.

 👍. Success:

  1. If the author meets requirements and the action is executed successfully, the reaction '👍' will be added to the comment.

File: .github/workflows/03-ci-lock.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - cron: "30 2 * * *"
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  lock:
+    name: 🔐 Lock closed issues and PRs
+    runs-on: ubuntu-latest
```
Comment (srealmoreno) [inline .github/workflows/03-ci-lock.yml:14]: #### Change Ubuntu-latest to Ubuntu-20.04

Using the latest version may generate an error in the future

```suggestion
    runs-on: ubuntu-20.04
```

File: .github/workflows/02-ci-stale.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - cron: "0 8 * * *"
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  stale:
+    name: 🧹 Clean up stale issues and PRs
+    runs-on: ubuntu-latest
```
Comment (srealmoreno) [inline .github/workflows/02-ci-stale.yml:14]: #### Change Ubuntu-latest to Ubuntu-20.04

Using the latest version may generate an error in the future

```suggestion
    runs-on: ubuntu-20.04
```

File: .github/workflows/03-ci-rebase-squash.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+#
+#  👎. Failed:
+#   1. If the author doesn't meet the requirements,
+#      the reaction '👎' will be added to the comment
+#      and the comment will be answered indicating
+#      which has no necessary requirements (See comment below).
+#
+#   2. If the autor meets requirements and the action fails,
+#      the reaction '👎' will be added to the comment.
+
```
Comment (srealmoreno) [inline .github/workflows/03-ci-rebase-squash.yml:32]: You forgot a number sign `#`

```suggestion
#
```

File: .github/workflows/03-ci-rebase-squash.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        id: check-permissions
+        run: >-
+          echo "::set-output name=permission::${{(
+            github.event.comment.user.id == github.event.repository.owner.id ||
+            github.event.comment.author_association == 'MEMBER' ||
+            contains(github.event.issue.assignees.*.id, github.event.comment.user.id) ||
+            github.event.comment.user.id == github.event.issue.user.id
+          )}}"
+      - uses: hmarr/debug-action@v2
+
```
Comment (srealmoreno) [inline .github/workflows/03-ci-rebase-squash.yml:72]: Remove debug

```suggestion
```

**Review by srealmoreno — COMMENTED** [review]
Rename **03-ci-rebase-squash.yml** to **04-ci-rebase-squash.yml**

**Review by OmarRizo — DISMISSED** [review]

File: .github/workflows/04-ci-rebase-squash.yml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+#   comment: "@gitbot rebase"  for rebase branch
+#   comment: "@gitbot squash"  for squash branch
+#
+# Requirements:
+#   The author of the comment must meet at least
+#   one requirement:
+#     - Be the owner of the repository
+#     - Be a member of the organization
+#     - Be on the assignees list
+#     - Be the author of the commit
```
Comment (srealmoreno) [inline .github/workflows/04-ci-rebase-squash.yml:17]: Did you mean `pull request`?
```suggestion
#     - Be the author of the pull request
```

File: .github/workflows/04-ci-rebase-squash.yml [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+# Interactions:
+#  👀. Caught comment:
+#   1. When the comment is detected (e.g. @gitbot rebase)
+#      the reaction '👀' will be added to the comment.
+#
+#  👎. Failed:
+#   1. If the author doesn't meet the requirements,
+#      the reaction '👎' will be added to the comment
+#      and the comment will be answered indicating
+#      which has no necessary requirements (See comment below).
```
Comment (srealmoreno) [inline .github/workflows/04-ci-rebase-squash.yml:28]: I think a better wording would be:
```suggestion
#      that it doesn't meet the necessary requirements.
#      (See comment below).
```

File: .github/workflows/04-ci-rebase-squash.yml [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+#
+#   2. If the autor meets requirements and the action fails,
```
Comment (srealmoreno) [inline .github/workflows/04-ci-rebase-squash.yml:30]: Did you mean `author`?

```suggestion
#   2. If the author meets requirements and the action fails,
```

File: .github/workflows/04-ci-rebase-squash.yml [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          comment-id: ${{ github.event.comment.id }}
+          reactions: "-1"
+
+  permission-denied:
+    needs: [catch-comment]
+    name: ⛔ Permission denied
+    if: needs.catch-comment.outputs.permission == 'false'
+    runs-on: ubuntu-22.04
```
Comment (srealmoreno) [inline .github/workflows/04-ci-rebase-squash.yml:109]: Change `ubuntu-22.04` to `ubuntu-20.04`
```suggestion
    runs-on: ubuntu-20.04
```

**Review by OmarRizo — APPROVED** [review]

**Review by OmarRizo — APPROVED** [review]

**Pull request merged**